File: back/src/LaMaps.Desktop/Gpx/Track.cs
using System.Collections.Generic;
using System.Collections.Immutable;
using System.Linq;
using NetTopologySuite.IO;

namespace LaMaps.Desktop.Controllers
{
    public class Track
    {
        public Track(GpxTrack gpxTrack)
        {
            Classification = gpxTrack.Classification;
            Comment = gpxTrack.Comment;
            Description = gpxTrack.Description;
            Name = gpxTrack.Name;
            Number = gpxTrack.Number;
            Source = gpxTrack.Source;
            Segments = gpxTrack.Segments
                .Select(s => new Segment()
                {
                    Waypoints = s.Waypoints.Select(w => new LaWaypoint(w)).ToList()
                }).ToImmutableList();
            Links = gpxTrack.Links;
            Extensions = (gpxTrack.Extensions as ImmutableXElementContainer).Items.Select(x =>
                    KeyValuePair.Create(x.Name.LocalName, x.Value))
                .ToImmutableList();
        }

        public ImmutableList<KeyValuePair<string, string>> Extensions { get; set; }

        public ImmutableArray<GpxWebLink> Links { get; set; }

        public uint? Number { get; set; }

        public ImmutableList<Segment> Segments { get; set; }

        public string Source { get; set; }

        public string Name { get; set; }

        public string Description { get; set; }

        public string Comment { get; set; }

        public string Classification { get; set; }
    }
}

File: front/src/api/models/Track.ts
/* istanbul ignore file */
/* tslint:disable */
/* eslint-disable */

import type { GpxWebLink } from './GpxWebLink';
import type { Segment } from './Segment';
import type { StringStringKeyValuePair } from './StringStringKeyValuePair';

export type Track = {
    extensions?: Array<StringStringKeyValuePair> | null;
    links?: Array<GpxWebLink>;
    number?: number | null;
    segments?: Array<Segment> | null;
    source?: string | null;
    name?: string | null;
    description?: string | null;
    comment?: string | null;
    classification?: string | null;
}